File: 2024/AdventOfCode/Tests/Day04Test.cs
using NUnit.Framework;

namespace AdventOfCode.Tests;

[TestFixture]
public class Day04Test {
    private const string Example = "Inputs/04-Example.txt";
    private const string Input = "Inputs/04.txt";
    private static Day04 _day;

    [TestFixture]
    public class Part1 {
        [Test]
        public void TestExample() {
            _day = new Day04(Example);
            const string expected = "18";
            var solution = _day.Solve_1();
            Assert.That(solution.Result, Is.EqualTo(expected));
        }

        [Test]
        public void TestInput() {
            _day = new Day04(Input);
            const string expected = "2549";
            var solution = _day.Solve_1();
            Assert.That(solution.Result, Is.EqualTo(expected));
        }
    }

    [TestFixture]
    public class Part2 {
        [Test]
        public void TestExample() {
            _day = new Day04(Example);
            const string expected = "9";
            var solution = _day.Solve_2();
            Assert.That(solution.Result, Is.EqualTo(expected));
        }

        [Test]
        public void TestInput() {
            _day = new Day04(Input);
            const string expected = "2003";
            var solution = _day.Solve_2();
            Assert.That(solution.Result, Is.EqualTo(expected));
        }
    }
}

File: 2025/test/advent_of_code/day_04_test.exs
defmodule AdventOfCode.Day04Test do
  use ExUnit.Case

  import AdventOfCode.Day04

  @day 4
  @year 2025

  @example String.trim("""
           ..@@.@@@@.
           @@@.@.@.@@
           @@@@@.@.@@
           @.@@@@..@.
           @@.@@@@.@@
           .@@@@@@@.@
           .@.@.@.@@@
           @.@@@.@@@@
           .@@@@@@@@.
           @.@.@@@.@.
           """)

  test "part1 example" do
    result = part1(@example)

    assert result == 13
  end

  test "part1" do
    input = AdventOfCode.Input.get!(@day, @year)
    result = part1(input)

    assert result == 1346
  end

  test "part2 example" do
    result = part2(@example)

    assert result == 43
  end

  test "part2" do
    input = AdventOfCode.Input.get!(@day, @year)
    result = part2(input)

    assert result == 8493
  end
end
